### PR TITLE
Feature/yaml output

### PR DESCRIPTION
**Pull request by MeCode4Food**
Added support for YAML and YML paths for the openapi middleware

e.g. openapi.yaml and openapi.yml

### CODE REVIEW

**Comment by MeCode4Food** [comment]
# #15

File: index.js [hunk 1]
```diff
@@ -134,6 +135,20 @@ module.exports = function ExpressOpenApi (_routePrefix, _doc, _opts) {
     middleware.document = generateDocument(middleware.document, req.app._router || req.app.router)
     res.json(middleware.document)
   })
+
+  // OpenAPI document as yaml
+  router.get(new RegExp(`${routePrefix}.[(yaml)(yml)]`), (req, res) => {
```
Comment (wesleytodd) [inline index.js:140]: Routes can also be specified as an array of paths, I would prefer this over the regexp approach.

```suggestion
  router.get([`${routePrefix}.yaml`, `${routePrefix}.yml`], (req, res) => {
```

File: index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const makeValidator = require('./lib/validate')
 const { get: getSchema, set: setSchema } = require('./lib/layer-schema')
 const minimumViableDocument = require('./lib/minimum-doc')
 const generateDocument = require('./lib/generate-doc')
+const convertYaml = require('./lib/convert-yaml')
```
Comment (wesleytodd) [inline index.js:10]: Lets just inline this below.

File: index.js [hunk 3]
```diff
@@ -134,6 +135,20 @@ module.exports = function ExpressOpenApi (_routePrefix, _doc, _opts) {
...
     res.json(middleware.document)
   })
+
+  // OpenAPI document as yaml
+  router.get(new RegExp(`${routePrefix}.[(yaml)(yml)]`), (req, res) => {
+    const jsonSpec = generateDocument(middleware.document, req.app._router || req.app.router)
+    const yamlSpec = convertYaml(jsonSpec)
+
+    // no need if req.originalUrl is undefined or not, will match prefix.yaml or prefix.yml
+    const docType = req.originalUrl.split('.')[1]
```
Comment (wesleytodd) [inline index.js:145]: This is not quite enough to make the approach work, but also I think a more simple `res.type('yaml')` would be enough.  This call maps to [this underlying lookup](https://github.com/jshttp/mime-db/blob/master/db.json#L7884-L7886), and both are aliases to the same content-type.

**Review by wesleytodd — CHANGES_REQUESTED** [review]
Thanks for this PR.  Overall it looks good, just a few small changes and I can get this merged.

**Comment by wesleytodd** [comment]
Oh, and it would also be helpful if you could squash this down to a single commit and follow the [conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/) for the commit message.

File: index.js (hunk 3 above)
Comment (MeCode4Food) [reply index.js]: That was weird, I tried it with `'yaml'` before and it didn't work.

**Comment by MeCode4Food** [comment]
I've squashed the commits with fixes in place. 

If the PR is alright, mind if I trouble you to add a `hacktoberfest-accepted` label to the PR?

File: index.js (hunk 3 above)
Comment (wesleytodd) [reply index.js]: Hm, what didn't work about it?

File: index.js (hunk 2 above)
Comment (wesleytodd) [reply index.js]: This one doesn't look like it has been resolved.  Can you just inline the call to the `yaml` package and remove this extra file?

File: index.js (hunk 3 above)
Comment (MeCode4Food) [reply index.js]: No, as in I tried it, it didn't work, but it works now. Thanks for the suggestion though!

File: index.js (hunk 2 above)
Comment (MeCode4Food) [reply index.js]: Yeah I'm not sure why is it now being reflected here, but I've included all the changes and now the index.js is shown to be "outdated". Its probably due to the way I squashed the commits

You can take a look at the "files changed" section of the PR for the changes I've made

Comment (wesleytodd) [reply index.js]: Yeah I see what you mean, but it still looks like the new file wasn't deleted, which is what I saw before making this comment.

File: lib/convert-yaml.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const YAML = require('yaml')
```
Comment (wesleytodd) [inline lib/convert-yaml.js:1]: This file is what I saw and commented on where it was imported.  Sorry for the confusion.

File: index.js (hunk 2 above)
Comment (wesleytodd) [reply index.js]: Hey @MeCode4Food, any chance you want to follow up with this? If not, I might just take over the PR myself next time I can work on this.  Just let me know if you would like to continue working on this.

File: lib/convert-yaml.js (hunk 4 above)
Comment (wesleytodd) [reply lib/convert-yaml.js:1]: The last thing to follow up on is to remove this file I think, since it was inlined where it is used.

**Pull request closed without merging**